test/common/utils/wait_for.js

```diff
@@ -205,6 +213,48 @@ function confirmations(transactions, limitHeight) {
 	return waitUntilLimit(limitHeight);
 }
 
+function confirmTransaction() {
+	return new Promise((resolve, reject) => {
+		nodeStatus((err, data) => {
+			if (err) {
+				return reject(err);
+			}
+			var total = data.transactions.total;
+			var target = data.transactions.confirmed;
+			var counter = 1;
+			async.doWhilst(
+				cb => {
+					nodeStatus((err, res) => {
+						if (err) {
+							cb(err);
+						}
+						__testContext.debug(
+							'	Waiting for transaction:'.grey,
+							'Total:'.grey,
+							res.transactions.total,
+							'Confirmed:'.grey,
+							res.transactions.confirmed,
+							'Second:'.grey,
+							counter++
+						);
+						target = res.transactions.confirmed;
+						setTimeout(cb, 1000);
+					});
+				},
+				() => {
+					return !(target >= total);
+				},
+				err => {
+					if (err) {
+						return reject(err);
+					}
+					return resolve(counter);
+				}
+			);
+		});
+	});
+}
+
```

Comment:
@ManuGowda @MaciejBaj This logic will suppress the actual cause of the problem and makes our tests false positives. 

We post a known N number of transactions and we should have a formula to calculate how many blocks required to confirm those. We can add some safety threshold on top of it. But still it would remain a definite number. 

Consider we mess up something in code e.g. `fillPool` that it don't pick correct number of transactions, so where we need 2 blocks it starts taking 10 blocks. This test will still pass in that case and we will never notice it again. 

I will not recommend merging this logic, instead keep failing nightly builds, until we identify and fix the root cause. So at least we will always have a negative but correct indicator.

Reply:
second

Reply:
@nazarhussain I agree we need to have a static number of the block to wait for to confirm the transaction on all nodes. And also we need to consider calculation differently for
1)  Syncing and broadcasting enabled
2)  Syncing disabled and broadcasting enabled
3)  Syncing enabled and broadcasting disabled

while running scenario 2 and 3, depending on block propagation and broadcasting mechanisum there will be possibilty of transaction not being confirmed on other nodes, so this will again cause the test to fail. At the moment we do not have a benchamark to determine how much block it takes to process 1000 transaction, so in order to avoid this fix we need to come up this the number for all the 3 scenarios and the network should work as expected for those number, until then the nightly builds can be kept failing.